src/pages/stageNames.tsx

```diff
@@ -1,75 +1,86 @@
 import React, { ReactElement } from 'react';
 import { NextPageContext } from 'next';
 import { parseCookies } from 'nookies';
-
+import _ from 'lodash';
 import TwoThirdsLayout from '../layout/Layout';
 import { FARE_STAGES_COOKIE, STAGE_NAMES_COOKIE, STAGE_NAME_VALIDATION_COOKIE } from '../constants';
 import { deleteCookieOnServerSide } from '../utils';
 import CsrfForm from '../components/CsrfForm';
-import { CustomAppProps } from '../interfaces';
+import { CustomAppProps, ErrorInfo } from '../interfaces';
+import FormElementWrapper from '../components/FormElementWrapper';
+import ErrorSummary from '../components/ErrorSummary';
 
 const title = 'Stage Names - Fares Data Build Tool';
 const description = 'Stage Names entry page of the Fares Data Build Tool';
 
 export interface InputCheck {
     error: string;
     input: string;
+    id: string;
 }
 
 type StageNameProps = {
     numberOfFareStages: number;
     inputChecks: InputCheck[];
+    errors: ErrorInfo[];
 };
 
-export const renderInputField = (index: number, inputCheck: InputCheck): ReactElement => (
+export const renderInputField = (index: number, inputCheck: InputCheck, errors: ErrorInfo[] = []): ReactElement => (
     <div className={`govuk-form-group${inputCheck?.error ? ' govuk-form-group--error input-error' : ''}`}>
```

Comment:
This might be worth fixing as well 

Each child in a list should have a unique "key" prop.

Check the render method of `StageNames`. See https://fb.me/react-warning-keys for more information.
    in div (at stageNames.tsx:29)

On line 29 if you add the unique key that should do the trick using {`fare-stage-name-${index + 1}`